typescript/sdk/src/providers/transactions/submitter/IcaTxSubmitter.ts

```diff
@@ -0,0 +1,203 @@
+import { ChainName } from '../../../types.js';
+import { MultiProvider } from '../../MultiProvider.js';
+import {
+  AnnotatedEV5Transaction,
+  ProtocolTypedReceipt,
+} from '../../ProviderType.js';
+import { CallData } from '../types.js';
+
+import { TxSubmitterInterface } from './TxSubmitterInterface.js';
+import { TxSubmitterType } from './TxSubmitterTypes.js';
+import { EV5GnosisSafeTxBuilder } from './ethersV5/EV5GnosisSafeTxBuilder.js';
+import { EV5GnosisSafeTxSubmitter } from './ethersV5/EV5GnosisSafeTxSubmitter.js';
+import { EV5ImpersonatedAccountTxSubmitter } from './ethersV5/EV5ImpersonatedAccountTxSubmitter.js';
+import { EV5JsonRpcTxSubmitter } from './ethersV5/EV5JsonRpcTxSubmitter.js';
+import { EvmIcaTxSubmitterPropsSchema } from './ethersV5/types.js';
+
+const EvmIcaTxSubmitterConfigSchema = EvmIcaTxSubmitterPropsSchema.required({
+  originInterchainAccountRouter: true,
+});
+
+/* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+const EvmIcaTxSubmitterConstructorConfigSchema =
+  EvmIcaTxSubmitterConfigSchema.omit({ internalSubmitter: true }).required({
+    owner: true,
+  });
+
+type EvmIcaTxSubmitterConfig = z.infer<typeof EvmIcaTxSubmitterConfigSchema>;
+type EvmIcaTxSubmitterConstructorConfig = z.infer<
+  typeof EvmIcaTxSubmitterConstructorConfigSchema
+>;
+
+type SubmitterFactoryMapping<
+  E extends TxSubmitterType,
+  TConfig extends { type: E },
+  TResult,
+> = {
+  [K in E]: (
+    config: Extract<TConfig, { type: K }>,
+  ) => Promise<TResult> | TResult;
+};
+
+async function getInternalSubmitter(
+  chain: ChainName,
+  multiProvider: MultiProvider,
+  config: EvmIcaTxSubmitterConfig['internalSubmitter'],
+): Promise<TxSubmitterInterface<ProtocolType.Ethereum>> {
+  const internalSubmitterMap: SubmitterFactoryMapping<
+    EvmIcaTxSubmitterConfig['internalSubmitter']['type'],
+    EvmIcaTxSubmitterConfig['internalSubmitter'],
+    TxSubmitterInterface<ProtocolType.Ethereum>
+  > = {
+    [TxSubmitterType.GNOSIS_SAFE]: (config) => {
+      return EV5GnosisSafeTxSubmitter.create(multiProvider, {
+        chain,
+        ...config,
+      });
+    },
+    [TxSubmitterType.GNOSIS_TX_BUILDER]: (config) => {
+      return EV5GnosisSafeTxBuilder.create(multiProvider, {
+        chain,
+        ...config,
+      });
+    },
+    [TxSubmitterType.IMPERSONATED_ACCOUNT]: (config) => {
+      return new EV5ImpersonatedAccountTxSubmitter(multiProvider, {
+        chain,
+        ...config,
+      });
+    },
+    [TxSubmitterType.JSON_RPC]: (config) => {
+      return new EV5JsonRpcTxSubmitter(multiProvider, {
+        chain,
+        ...config,
+      });
+    },
+  };
+
+  const internalSubmitterFactory = internalSubmitterMap[config.type];
+  // Sanity check
+  if (!internalSubmitterFactory) {
+    throw new Error(
+      `Internal submitter factory not found for type: ${config.type}`,
+    );
+  }
+
+  return internalSubmitterFactory(config as any);
+}
```

Comment:
ditto here
this doesnt seem like it should be necessary for the ICA submitter to implement